shopfloor/services/delivery.py

```diff
@@ -104,17 +386,63 @@ def set_qty_done_line(self, picking_id, line_id):
         Transitions:
         * deliver: always return here with updated data
         """
-        return self._response()
+        picking = self.env["stock.picking"].browse(picking_id).exists()
+        if picking:
+            response = self._check_picking_status(picking)
+            if response:
+                return response
+        else:
+            return self._response_for_deliver(
+                message=self.msg_store.stock_picking_not_found()
+            )
+        line = self.env["stock.move.line"].browse(move_line_id).exists()
+        if line:
+            if line.package_id:
+                return self._response_for_deliver(
+                    picking=picking,
+                    message=self.msg_store.line_has_package_scan_package(),
+                )
+            self._set_lines_done(line)
+            self._action_picking_done(picking)
+            return self._response_for_deliver(picking)
+        return self._response_for_deliver(
+            picking=picking, message=self.msg_store.record_not_found(),
+        )
 
     def reset_qty_done_pack(self, picking_id, package_id):
         """Remove "Done" on a package
 
         Transitions:
         * deliver: always return here with updated data
         """
-        return self._response()
+        picking = self.env["stock.picking"].browse(picking_id).exists()
+        if picking:
+            response = self._check_picking_status(picking)
+            if response:
+                return response
+        else:
+            return self._response_for_deliver(
+                message=self.msg_store.stock_picking_not_found()
+            )
+        package = self.env["stock.quant.package"].browse(package_id).exists()
+        if package:
+            lines = self.env["stock.move.line"].search(
+                self._lines_from_package_domain(package, no_qty_done=False)
+            )
+            if not lines:
+                return self._response_for_deliver(
+                    picking,
+                    message=self.msg_store.package_not_available_in_picking(
+                        package, picking
+                    ),
+                )
+            self._reset_lines(lines)
+            return self._response_for_deliver()
```

Comment:
always return picking when possible :pray: 
```suggestion
            return self._response_for_deliver(picking=picking)
```